### ISSUE TITLE
Does the gradient salience map support pytorch model

### ISSUE THREAD

**Issue by wmmxk** [PAIR-code/lit#19]
I saw in the ```predict_minibatch()``` method of the example model, [tf.GradientTape ](https://github.com/PAIR-code/lit/blob/80f4f771a93ebd6c294f29d827f3993bc830958b/lit_nlp/examples/models/glue_models.py#L230)is used to restore the gradient. Does this mean pytorch model is not supported for now?

**Comment by iftenney** [comment]
LIT treats the gradient tape as internal to the model, and just looks for the `TokenGradients` field in the output (which should contain NumPy arrays). So for PyTorch, you'd just want the equivalent of this line: https://github.com/PAIR-code/lit/blob/80f4f771a93ebd6c294f29d827f3993bc830958b/lit_nlp/examples/models/glue_models.py#L259 using e.g. `torch.autograd.grad`

**Comment by wmmxk** [comment]
@iftenney Thanks for your reply. I see ```torch.autograd.grad``` support calculating the gradient of a variable with respect to another variable. I can specify the two variables as you did in ```[batched_outputs["input_emb_grad"] = tape.gradient(scalar_pred_for_gradients, embs[0])](https://github.com/PAIR-code/lit/blob/80f4f771a93ebd6c294f29d827f3993bc830958b/lit_nlp/examples/models/glue_models.py#L259)```. 

I am not sure there is other details I need pay special attention. I will update later in this thread.

**Comment by wmmxk** [comment]
For a T5 model, the output is created in a generative manner. I was struggling with calculating the gradient regardless of what language to use, Tensorflow or Pytorch. Any ideas?

**Comment by LittlePea13** [comment]
You can check this PR, https://github.com/PAIR-code/lit/pull/46, where there is an example on how to get gradients with Pytorch.

**Comment by iftenney** [comment]
Fixed with #46
